README.md

```diff
@@ -1,4 +1,172 @@
+  <tr>
+    <th rowspan="2">Dataset name</th>
+    <th rowspan="2">Link</th>
+    <th colspan="3">Number of projects</th>
+    <th colspan="3">Number of functions</th>
+  </tr>
+  <tr>
+    <td>train</td>
+    <td>val</td>
+    <td>test</td>
+    <td>train</td>
+    <td>val</td>
+    <td>test</td>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>Java-small</td>
+    <td><a href="https://s3-eu-west-1.amazonaws.com/datasets.ml.labs.aws.intellij.net/java-ast-methods/java-ast-methods-small.tar.gz">s3 link</a></td>
+    <td>10</td>
+    <td>1</td>
+    <td>1</td>
+    <td>780 970</td>
+    <td>32 654</td>
+    <td>59 381</td>
+  </tr>
+  <tr>
+    <td>Java-medium</td>
+    <td><a href="https://s3-eu-west-1.amazonaws.com/datasets.ml.labs.aws.intellij.net/java-ast-methods/java-ast-methods-medium.tar.gz">s3 link</a></td>
+    <td>800</td>
+    <td>100</td>
+    <td>96</td>
+    <td>3 341 148</td>
+    <td>457 527</td>
+    <td>450 260</td>
+  </tr>
+  <tr>
+    <td>Java-large</td>
+    <td><a href="https://s3-eu-west-1.amazonaws.com/datasets.ml.labs.aws.intellij.net/java-ast-methods/java-ast-methods-large.tar.gz">s3 link</a></td>
+    <td>8999</td>
+    <td>250</td>
+    <td>307</td>
+    <td>16 794 775</td>
+    <td>~350 000</td>
+    <td>459 104</td>
+  </tr>
+</tbody>
+</table>
+
+## Tree2Seq framework
+
+Implementation based on [PyTorch](https://pytorch.org/docs/stable/torch.html) framework.
+For supporting learning on graphs [DGL](https://www.dgl.ai/) library is used.
+
+Framework consist of 3 main parts:
+- Node embedding (`INodeEmbedding`) that embed tokens in nodes to vectors.
+- Tree encoder (`ITreeEncoder`) that encode tree into a vector representation.
+- Tree decoder (`ITreeDecoder`) that decode vector representation into target sequence.
+
+Converting source tree into target sequence occurs in `Tree2Seq` class and represent sequence applying of interface realization.
+If you want to add a new algorithm, you should realize class with inheritance from the needed interface, and then use it in the configuration.
+
+### Configuration
+
+Components configuration and required hyper parameters can be passed to the model throw `.json` files.
+The examples of such files can be found in `config` directory, there are already prepared configs for training and evaluating
+Tree-LSTM model on Java dataset.
+
+Use can use prepared scripts to interact with model:
+- Start training ChildSum Tree-LSTM with logging to [wandb](https://www.wandb.com/) service
+```bash
+python train.py configs/tree_lstm_childsum_java_small.json wandb
+```
+- Evaluate model:
+```bash
+python evaluate.py configs/evaluate.json
+```
+- Predict target sequence for given file with concrete model (predict function name by its body),
+this step includes data preprocessing and model applying:
+```bash
+python interactive.py test.java model.pt
+```
+
+## Evaluation
+
+Currently, there are implemented Tree-LSTM model with various modifications. For all of them, LSTM with Attention to subtrees
```

Comment:
Currently, several Tree-LSTM models with various modifications are implemented. For all of them, LSTM with Attention to subtrees is used to decode the representation into a target sequence (a method name). As for the node embedding, the matrix embeddings for types (internal nodes) and tokens (leafs) are employed.